src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -563,15 +563,20 @@ private string GetAbortErrorMessage(Exception exception, bool getClientError)
             {
                 EqtTrace.Verbose("TestRequestSender: GetAbortErrorMessage: Client has disconnected. Wait for standard error.");
 
-                // Set a default message and wait for test host to exit for a moment
-                reason = CommonResources.UnableToCommunicateToTestHost;
+                // Wait for test host to exit for a moment
                 if (this.clientExited.Wait(this.clientExitedWaitTime))
                 {
+                    // Set a default message of test host process exited and additionally specify the error if present
                     EqtTrace.Info("TestRequestSender: GetAbortErrorMessage: Received test host error message.");
-                    reason = this.clientExitErrorMessage;
+                    reason = CommonResources.TestHostProcessExitedDuringExecution;
+                    if (!string.IsNullOrWhiteSpace(this.clientExitErrorMessage))
+                    {
+                        reason = $"{reason} : {this.clientExitErrorMessage}";
```

Comment:
should it be?
reason = this.clientExitErrorMessage

Current code would look bit odd.
The active test run was aborted. Reason: Test Host process exited during the execution of tests. : <actual error message>

Can we add a test for this?

Reply:
I wanted the user to be aware that the abort was due to a Test Host Process exit.. will remove the execution of tests as the same message is used in a discovery scenario as well...

---
In reply to: [266397418](https://github.com/Microsoft/vstest/pull/1961#discussion_r266397418) [](ancestors = 266397418)

Reply:
There are existing tests covering this.. will fix them 

---
In reply to: [266768756](https://github.com/Microsoft/vstest/pull/1961#discussion_r266768756) [](ancestors = 266768756,266397418)